### PR TITLE
chore(vue-app): suppress deprecated warning for production

### LINKED ISSUE #5136: Warning is shown in built bundle (Classic mode for store/ is deprecated and will be removed in Nuxt 3).
### Version

[v2.4.5](https://github.com/nuxt.js/releases/tag/v2.4.5)

### Reproduction link

[https://codesandbox.io/s/jjyz70wpm3](https://codesandbox.io/s/jjyz70wpm3)

### Steps to reproduce

N/A

### What is expected ?

No warning in built bundle.

### What is actually happening?

Warning "Classic mode for store/ is deprecated and will be removed in Nuxt 3" is shown for https://nuxtjs.org/guide/vuex-store/#classic-mode.

### Additional comments?

How can I prevent showing this warn?

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8756">#c8756</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by sugarshin**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Fixes #5136 
We should not need this warning in the production build.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by sugarshin** [comment]
Security Audit is failed. Is there anything I can do for that?
https://circleci.com/gh/nuxt/nuxt.js/31459?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by manniL** [comment]
Hey 👋 Thanks for the contribution Left you some comments :)

File: packages/vue-app/template/store.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ void (function updateModules() {
   <% return true }}) %>
 
   // If store is an exported method = classic mode (deprecated)
-  if (typeof store === 'function') {
+  if (typeof store === 'function' && process.env.NODE_ENV !== 'production') {
```
Comment (manniL) [inline packages/vue-app/template/store.js]: Better use `<% if (isDev) { %>` so it's not included in the bundle itself ☺️

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

File: packages/vue-app/template/store.js (hunk 1 above)
Comment (sugarshin) [reply packages/vue-app/template/store.js]: Thank you for your feedback! I fixed it!

**Review by TheAlexLichter — DISMISSED** [review]

**Comment by manniL** [comment]
LGTM ☺️

**Comment by sugarshin** [comment]
Thank you 🙏 please merge it.

**Review by pi0 — APPROVED** [review]

**Pull request merged**